components/support/webextensions/src/main/java/mozilla/components/support/webextensions/WebExtensionSupport.kt

```diff
@@ -189,6 +187,25 @@ object WebExtensionSupport {
             }
     }
 
+    /**
+     * Marks the provided [updatedExtension] as updated in the [store].
+     */
+    fun markExtensionAsUpdated(store: BrowserStore, updatedExtension: WebExtension) {
+        installedExtensions[updatedExtension.id] = updatedExtension
+        store.dispatch(WebExtensionAction.UpdateWebExtension(updatedExtension.toState()))
+
+        // Register action handler for all existing engine sessions on the new extension
```

Comment:
nit: on the updated extension. An issue was filed today to get us an API so we don't have to do this per extension: https://bugzilla.mozilla.org/show_bug.cgi?id=1603559

We could link to it here and in the code above.

Reply:
Great, I'll add it!